api/integrations/launch_darkly/services.py

```diff
@@ -165,6 +175,105 @@ def _create_string_feature_states(
         )
 
 
+# Based on: https://docs.launchdarkly.com/sdk/concepts/flag-evaluation-rules#operators
+def _ld_operator_to_flagsmith_operator(ld_operator: str) -> Optional[str]:
+    return {
+        "in": constants.IN,
+        "endsWith": constants.REGEX,
+        "startsWith": constants.REGEX,
+        "matches": constants.REGEX,
+        "contains": constants.CONTAINS,
+        "lessThan": constants.LESS_THAN,
+        "lessThanOrEqual": constants.LESS_THAN_INCLUSIVE,
+        "greaterThan": constants.GREATER_THAN,
+        "greaterThanOrEqual": constants.GREATER_THAN_INCLUSIVE,
+        "before": constants.LESS_THAN,
+        "after": constants.GREATER_THAN,
+        "semVerEqual": constants.EQUAL,
+        "semVerLessThan": constants.LESS_THAN,
+        "semVerGreaterThan": constants.GREATER_THAN,
+        "segmentMatch": constants.REGEX,
+    }.get(ld_operator, None)
+
+
+# TODO: Not sure what happens if it is not `IN` and there are multiple values.
+def _convert_ld_value(values: list[str], ld_operator: str) -> str:
+    match ld_operator:
+        case "in":
+            return ",".join(values)
+        case "endsWith":
+            return ".*" + re.escape(values[0])
+        case "startsWith":
+            return re.escape(values[0]) + ".*"
+        case "matches" | "segmentMatch":
+            return re.escape(values[0]).replace("\\*", ".*")
+        case "contains":
+            return ".*" + re.escape(values[0]) + ".*"
+        case _:
+            return values[0]
+
+
+def _clauses_to_segment(
+    clauses: list[Clause],
+    project: Project,
+    feature: Feature,
+    environment: Environment,
+    name: str,
+) -> Segment:
+    segment = Segment.objects.create(name=name, project=project, feature=feature)
+
+    # A parent rule has two children: one for the regular conditions and multiple for the negated conditions
+    # Mathematically, this is equivalent to:
+    # any(X, Y, !Z) = any(any(X,Y), none(Z))
+    # Or with more parameters,
+    # any(X, Y, !Z, !W) = any(any(X,Y), none(Z), none(W))
+    # Since there is no !X operation in Flagsmith, we wrap negated conditions in a none() rule.
+    parent_rule = SegmentRule.objects.create(segment=segment, type=SegmentRule.ANY_RULE)
+    child_rule = SegmentRule.objects.create(rule=parent_rule, type=SegmentRule.ANY_RULE)
+
+    for clause in clauses:
+        operator = _ld_operator_to_flagsmith_operator(clause["op"])
+        value = _convert_ld_value(clause["values"], clause["op"])
+        _property = clause["attribute"]
+
+        if operator is not None:
+            if clause["negate"] is True:
+                negated_child = SegmentRule.objects.create(
+                    rule=parent_rule, type=SegmentRule.NONE_RULE
+                )
+                rule = negated_child
+            else:
+                rule = child_rule
```

Comment:
@khvn26 this block needs some attention 🙂

Reply:
Also LMK if there are any other testing strategies than manual testing. Probably we can download the JSON responses from a test project and try to mock it here.

Reply:
We cover all of our code with unit tests. Refer to [existing LaunchDarkly importer unit tests](https://github.com/Flagsmith/flagsmith/blob/93df958b8fc67d17c6cfc298184bfef0f83847b4/api/tests/unit/integrations/launch_darkly) to see how mocking is organised.

Reply:
Regarding the code block in question, yeah, I was about to suggest using the `NONE_RULE` rule type for the negated LD clauses. Looks good but I expect it to be thoroughly covered with tests!

Reply:
Kinda verified this by checking the frontend after import, they are shown as `AND NOT`.

One UI issue I identified is first rule that is `NONE` is not properly shown in the UI. Probably we should show like a header that says `---- Not ----`?

wdy think?

Reply:
The display bug is now fixed.